tensorflow_addons/losses/metric_learning.py

```diff
@@ -64,3 +64,27 @@ def pairwise_distance(feature, squared=False):
     )
     pairwise_distances = tf.math.multiply(pairwise_distances, mask_offdiagonals)
     return pairwise_distances
+
+
+@tf.function
+def angular_distance(feature):
+    """Computes the angular distance matrix
+
+    output[i, j] = 1 - cosine_similarity(feature[i, :], feature[j, :])
+
+    Args:
+      feature: 2-D Tensor of size [number of data, feature dimension].
+
+    Returns:
+      angular_distances: 2-D Tensor of size [number of data, number of data].
+    """
+    # normalize input
+    feature = tf.math.l2_normalize(feature, axis=1)
+
+    # create adjaceny matrix of cosine similarity
+    angular_distances = 1 - tf.matmul(feature, tf.transpose(feature))
```

Comment:
`tf.matmul` has the option to transpose matrix (`transpose_b`) so you do not need to manually transpose it.

https://www.tensorflow.org/api_docs/python/tf/linalg/matmul#expandable-1-label